db/src/db.ts

```diff
@@ -145,7 +145,7 @@ export class DB extends EventEmitter {
    */
   public async create(key: string, value: Serializable): Promise<boolean> {
```

Comment:
This function no longer needs to be `async`.  What do you (we) think about that?  (You mentioned perf, so AFAIU this _does_ create an extra `Promise`.  I don't know if the JITs don't strip them out, obviously.)
```
> async function af() { return 17; }
undefined
> af()
Promise {
  17,
  domain:
   Domain {
     domain: null,
     _events:
      [Object: null prototype] {
        removeListener: [Function: updateExceptionCapture],
        newListener: [Function: updateExceptionCapture],
        error: [Function: debugDomainError] },
     _eventsCount: 3,
     _maxListeners: undefined,
     members: [],
     [Symbol(kWeak)]: WeakReference {} } }
```

Reply:
> There is a subtle difference in unmaking it `async` - if `checkValue` throws it's going to be an exception rather than a rejection. Usually it doesn't matter that much, but should not be changed without checking.

I agree that there's a difference, and actually I think it's an improvement (thanks!).  I don't think that we depend on the existing behaviour.  And I *do* believe that this code is (potentially) useful -- making the "sync promise" style the most expressive for something like this:
```ts
function setVals(vs: any[]) {
  try {
    return vs.map((val, idx) => create(`k${idx}`, val));
  } catch (e) {
    if (...) throw new TypeError('some value was bad');
  }
}
function waitAndSet(vs: any[]) {
  try {
    await Promise.all(setVals());
    return true;
  } catch (e) {
    if (e instanceof TypeError) throw e;
    log('bad db', e);
    return false
  }
}
```

If you agree that this style is best, can we just specify it *now*, so that we never have to change it later?